lib/SPIRV/libSPIRV/SPIRVModule.cpp

```diff
@@ -1618,10 +1622,13 @@ class TopologicalSort {
     for (auto ES : EntryStateMap)
       visit(ES.first);
   }
+
+  SPIRVFunctionVec& getFunctionVec() { return FunctionVec; }
 };
 
 spv_ostream &operator<<(spv_ostream &O, const TopologicalSort &S) {
-  O << S.TypeIntVec << S.ConstIntVec << S.TypeVec << S.ConstAndVarVec;
+  O << S.TypeIntVec << S.ConstIntVec << S.TypeVec << S.ConstAndVarVec
+    << S.FunctionVec;
```

Comment:
So if I understand correctly, you are now emitting all functions last?  It would be good to mention that in the commit message.

Reply:
Yes all ctor/dtor functions at last. Previously they belong to ConstAndVarVec.
Should I emit FunctionVec before ConstAndVarVec?

Reply:
The order is fine with your change I believe.  I only commented because I think it is worth mentioning in the commit message.

Reply:
Done. I've changed logic and emit ctor/dtor along with other functions.